model/jpa/src/main/resources/META-INF/jpa-changelog-9.0.0.xml

```diff
@@ -64,4 +64,10 @@
         <addPrimaryKey columnNames="USER_SESSION_ID, CLIENT_ID, CLIENT_STORAGE_PROVIDER, EXTERNAL_CLIENT_ID, OFFLINE_FLAG" constraintName="CONSTRAINT_OFFL_CL_SES_PK3" tableName="OFFLINE_CLIENT_SESSION"/>
     </changeSet>
 
+    <changeSet author="keycloak" id="9.0.0-KEYCLOAK-12579">
+        <!-- Now the parent group cannot be NULL to make SIBLING_NAMES unique constraint work -->
+        <!-- Top level groups are now marked with the empty string -->
+        <addNotNullConstraint tableName="KEYCLOAK_GROUP" columnName="PARENT_GROUP" columnDataType="VARCHAR(36)" defaultNullValue=""/>
```

Comment:
Ah! I didn't know that oracle behavior, I thought that it won't work in Oracle for sure. You can just test the GroupTest and tell me, because I think a lot of tests are going to fail. The code is more or less prepared to change the top name to anything, I'll use " " as you suggest.

Reply:
Sent you an email with some details about the results of DB run. The PR will need some work...

Reply:
Buf! OK, I'll rebase and test again.

Reply:
OK! Rebased and the top id is changed to " ". The issue with sql server is that in order to add a NOT NULL constraint the previous SIBLING_NAMES constraint should be dropped before and re-created after (I've realized that it's done this way in other XML files for other table changes). Let's see how the tests go.

Reply:
@rmartinc Travis failed in cross-dc tests, but it is unrelated to this PR. I re-run the pipeline tests for this. I hope to let you know around tomorrow. Thanks!